library/std/src/sys/unix/os.rs

```diff
@@ -586,6 +586,36 @@ pub fn temp_dir() -> PathBuf {
     crate::env::var_os("TMPDIR").map(PathBuf::from).unwrap_or_else(|| {
         if cfg!(target_os = "android") {
             PathBuf::from("/data/local/tmp")
+        } else if cfg!(target_vendor = "apple") {
+            extern "C" {
+                fn confstr(
+                    name: libc::c_int,
+                    buf: *mut libc::c_char,
+                    len: libc::size_t,
+                ) -> libc::size_t;
+            }
+            let tmpdir = unsafe { libc::getenv(b"TMPDIR".as_ptr() as *const libc::c_char) };
+            if tmpdir.is_null() {
+                let mut buf: Vec<u8> = vec![0; libc::PATH_MAX as usize];
```

Comment:
50 bytes should be enough in practice, but really this should have some retry logic. I was going to submit https://github.com/thomcc/rust/commit/ee1c648d4be0b9c23399450578d434018db8868b as a PR at one point, but you can just integrate the changes from it.

Reply:
it s better if you do since you authored all.